examples/helper_macros.rs

```diff
@@ -0,0 +1,27 @@
+//! A simple demonstration how iron's helper macros make e.g. IO-intensive code easier to write.
+#[macro_use] extern crate iron;
+
+use std::io;
+use std::fs;
+
+use iron::prelude::*;
+use iron::status;
+use iron::method;
+
+fn main() {
+    Iron::new(|req: &mut Request| {
+        Ok(match req.method {
+            method::Get => {
+                let f = itry!(fs::File::open("foo.txt"), (status::Ok, ""));
+                Response::with(status::Ok).set(f)
```

Comment:
It exists now so that if you pass a Path the content-type will be set automatically.